src/operator/regression_output.cc

```diff
@@ -106,13 +135,40 @@ It is suitable for binary classification or probability prediction tasks.
 .. note::
    Use the LogisticRegressionOutput as the final output layer of a net.
 
-By default, gradients of this loss function are scaled by factor `1/n`, where n is the number of training examples.
-The parameter `grad_scale` can be used to change this scale to `grad_scale/n`.
+By default, gradients of this loss function are scaled by factor `1/m`, where m is the number of dimensions of a training example.
+The parameter `grad_scale` can be used to change this scale to `grad_scale/m`.
 
 )code" ADD_FILELINE)
+.set_num_inputs(2)
+.set_num_outputs(1)
+.set_attr<nnvm::FListInputNames>("FListInputNames",
+  [](const NodeAttrs& attrs) {
+    return std::vector<std::string>{"data", "label"};
+  })
+.set_attr<nnvm::FInferShape>("FInferShape", RegressionOpShape)
+.set_attr<nnvm::FGradient>("FGradient", RegressionOpGrad{"_backward_logistic_reg_out"})
+.set_attr<nnvm::FInplaceOption>("FInplaceOption",
+[](const NodeAttrs& attrs){
+  return std::vector<std::pair<int, int> >{{0, 0}};
+})
+.set_attr<FCompute>("FCompute<cpu>", RegressionForward<cpu, mshadow_op::sigmoid>)
 .add_argument("data", "NDArray-or-Symbol", "Input data to the function.")
 .add_argument("label", "NDArray-or-Symbol", "Input label to the function.")
 .add_arguments(RegressionOutputParam::__FIELDS__());
 
+NNVM_REGISTER_OP(_backward_logistic_reg_out)
+.set_num_inputs(2)
+.set_num_outputs(2)
+.set_attr_parser(ParamParser<RegressionOutputParam>)
+.set_attr<nnvm::TIsBackward>("TIsBackward", true)
+.set_attr<nnvm::FInplaceOption>("FInplaceOption",
+[](const NodeAttrs& attrs){
+  // inputs are in_label and out_data, outputs are data_grad and label_grad
+  return std::vector<std::pair<int, int> >{{1, 0}};
+})
+.set_attr<FCompute>("FCompute<cpu>", RegressionBackward<cpu, mshadow_op::minus>);
```

Comment:
We can consider creating a macro (like in https://github.com/apache/incubator-mxnet/blob/master/src/operator/tensor/elemwise_binary_op.h#L554-L569) for defining regression OPs as most of the codes are similar.

Reply:
+1